chromium_src/third_party/blink/renderer/core/frame/location.cc

```diff
@@ -20,11 +20,18 @@ DOMStringList* Location::ancestorOrigins() const {
     return origin->Host().EndsWith(".onion", kTextCaseASCIIInsensitive);
   };
 
+  auto is_chrome_untrusted = [](const SecurityOrigin* origin) {
+    return origin->Protocol() == "chrome-untrusted";
+  };
+
   auto* raw_origins = ancestorOrigins_ChromiumImpl();
   for (uint32_t i = 0; i < raw_origins->length(); ++i) {
     const String raw_origin = raw_origins->item(i);
     const scoped_refptr<SecurityOrigin> origin =
         SecurityOrigin::CreateFromString(raw_origin);
+    if (is_chrome_untrusted(origin.get())) {
+      break;
```

Comment:
Sorry for missing that during the review, but shouldn't we use `continue` here instead?
I think we want to remove `chrome-untrusted` from the list, but if there are multiple nesting levels, then we should probably leave the other ones there.

Reply:
> Sorry for missing that during the review, but shouldn't we use `continue` here instead? I think we want to remove `chrome-untrusted` from the list, but if there are multiple nesting levels, then we should probably leave the other ones there.

we explicitly don't want the player iframe to know its nested level (how many iframes deep it's running).

I don't think we want to reveal this info to any iframe inside `chrome-untrusted://`.

Reply:
Just to be clear, I was not suggesting adding `"null"` entries in this case, just omitting the `chrome-untrusted` entries instead.

Reply:
> Just to be clear, I was not suggesting adding `"null"` entries in this case, just omitting the `chrome-untrusted` entries instead.

yep, but this way the `ancestorOrigins` will still contain the top frame, which is `chrome://player`. We also don't want this.

Reply:
And it would be fine to remove `chrome://` URLs in a similar way IMO.

So basically I think we should do one of two things:

1. Remove `chrome-untrusted` and `chrome` untrusted entries from `ancestorOrigins` in all cases, but leave the other entries alone. This would apply all of the time.
2. Clear `ancestorOrigins` entirely, but only for `chrome://player`.

With #1, it would mean that if we have nested iframes like:

     chrome://player => chrome-untrusted://whatever => https://example.com => https://targetsite.com

then `targetsite.com` would see the same `ancestorOrigins` as in the normal web case:

    https://example.com => https://targetsite.com

#2 on the other hand would solve the immediate problem without affecting anything else.

Reply:
> With #1, it would mean that if we have nested iframes like:
> 
> ```
>  chrome://player => chrome-untrusted://whatever => https://example.com => https://targetsite.com
> ```
> then targetsite.com would see the same ancestorOrigins as in the normal web case:
> ```
> https://example.com => https://targetsite.com
> ```

This example should already work like you described. The chain will be left intact until `chrome-untrusted://` is encountered, so `https://example.com => https://targetsite.com` chain will stay. The order in the chain list is reversed: first we get `https://targetsite.com`, then `https://example.com`, then `chrome-untrusted://`. We stop at `chrome-untrusted://` item, this means the "good" `https://` chain is there.